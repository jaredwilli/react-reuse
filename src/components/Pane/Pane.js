import React from 'react';
import PropTypes from 'prop-types';

/** Pane component for the atom of a tabs component */
const Pane = props => (
    <div className="tab-pane">
        {props.children}
    </div>
);

Pane.propTypes = {
    /** The label of the pane */
    label: PropTypes.string.isRequired,

    /** The children or content of the pane */
    children: PropTypes.element.isRequired
};

export default Pane;
