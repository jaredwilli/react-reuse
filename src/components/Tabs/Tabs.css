
.tabs {

}

.tabs .tabs__labels {
    border-bottom: 1px solid #29292a;
    list-style-type: none;
    margin: 0;
    padding: 0 0 0 30px;
}

.tabs .tabs__labels li {
    background: #fff;
    border: 1px solid #ccc;
    border-bottom-color: #29292a;
    text-align: center;
    display: inline-block;
    width: 110px;
    padding: 10px;
    position: relative;
    top: 1px;
    cursor: pointer;
}

.tabs .tabs__labels li.active {
    border: 1px solid #29292a;
    border-bottom-color: #fff;
}

.tabs .tabs__labels li:first-child {
    border-radius: 2px 0 0 0;
}

.tabs .tabs__labels li:last-child {
    border-radius: 0 2px 0 0;
}

.tabs .tabs__content {
    padding: 15px;
}
