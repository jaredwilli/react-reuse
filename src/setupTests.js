import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';

import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';

Enzyme.configure({ adapter: new Adapter() });

// Make Enzyme functions available in all test files without importing
global.shallow = shallow;
global.render = render;
global.mount = mount;
