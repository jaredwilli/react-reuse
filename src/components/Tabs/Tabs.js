import React from 'react';
import PropTypes from 'prop-types';

import './Tabs.css';

const Tabs = class extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            selected: props.selected
        };
    }

    _renderTitles = () => {
        function _labels(child, index) {
            const activeClass = this.state.selected === index
                ? 'active'
                : '';

            return (
                <li
                    key={index}
                    className={activeClass}
                    onClick={e => this.handleClick(index, e)}
                >
                    {child.props.label}
                </li>
            );
        }

        return (
            <ul className="tabs__labels">
                {this.props.children.map(_labels.bind(this))}
            </ul>
        );
    }

    _renderContent = () => {
        return (
            <div className="tabs__content">
                {this.props.children[this.state.selected]}
            </div>
        );
    }

    handleClick = (index, event) => {
        event.preventDefault();

        this.setState({
            selected: index
        });
    }

    render() {
        return (
            <div className="tabs">
                {this._renderTitles()}
                {this._renderContent()}
            </div>
        );
    }
};

Tabs.propTypes = {
    /** The selected tab number as an index */
    selected: PropTypes.number,

    /** The content of the selected tab pane */
    children: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.element
    ]).isRequired
};

Tabs.defaultProps = {
    selected: 0
};

export default Tabs;
