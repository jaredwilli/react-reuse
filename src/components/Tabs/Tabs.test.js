import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import * as setupTests from '../../setupTests';

import Tabs from './Tabs';
import Pane from '../Pane/Pane';

describe('Tabs', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<Tabs selected={0}><Pane label="Tab 1"><div>This is my tab 1 contents!</div></Pane><Pane label="Tab 2"><div>This is my tab 2 contents!</div></Pane><Pane label="Tab 3"><div>This is my tab 3 contents!</div></Pane></Tabs>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
