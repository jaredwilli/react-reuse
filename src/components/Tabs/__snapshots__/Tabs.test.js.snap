// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders the component correctly 1`] = `
<div
  className="tabs"
>
  <ul
    className="tabs__labels"
  >
    <li
      className="active"
      onClick={[Function]}
    >
      Tab 1
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      Tab 2
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      Tab 3
    </li>
  </ul>
  <div
    className="tabs__content"
  >
    <div
      className="tab-pane"
    >
      <div>
        This is my tab 1 contents!
      </div>
    </div>
  </div>
</div>
`;
